app/metrics.py

```diff
@@ -43,29 +43,29 @@ def get_previous_metrics(w3, pool, oracle, beacon_spec, from_block=0) -> PoolMet
+def get_light_current_metrics(w3, beacon, pool, oracle, beacon_spec):
     """If the result of previous get_current_metrics call isn't given
-    create and return partial metric.mSince it doesn't get keys from
-    registry and doesn't retrieve beacon state, it's much faster."""
-    if partial_metrics is None:
-        epochs_per_frame = beacon_spec[0]
-        partial_metrics = PoolMetrics()
-        partial_metrics.blockNumber = w3.eth.getBlock('latest')['number']  # Get the the epoch that is both finalized and reportable
-        current_frame = oracle.functions.getCurrentFrame().call()
-        potentially_reportable_epoch = current_frame[0]
-        logging.info(f'Potentially reportable epoch: {potentially_reportable_epoch} (from ETH1 contract)')
-        finalized_epoch_beacon = beacon.get_finalized_epoch()
-        # For Web3 client
-        # finalized_epoch_beacon = int(beacon.get_finality_checkpoint()['data']['finalized']['epoch'])
-        logging.info(f'Last finalized epoch: {finalized_epoch_beacon} (from Beacon)')
-        partial_metrics.epoch = min(potentially_reportable_epoch,
-                        (finalized_epoch_beacon // epochs_per_frame) * epochs_per_frame)
-        partial_metrics.timestamp = get_timestamp_by_epoch(beacon_spec, partial_metrics.epoch)
-        partial_metrics.depositedValidators = pool.functions.getBeaconStat().call()[0]
-        partial_metrics.bufferedBalance = pool.functions.getBufferedEther().call()
-        return partial_metrics
-
+     create and return partial metric.mSince it doesn't get keys from
+     registry and doesn't retrieve beacon state, it's much faster."""
+    epochs_per_frame = beacon_spec[0]
+    partial_metrics = PoolMetrics()
+    partial_metrics.blockNumber = w3.eth.getBlock('latest')['number']  # Get the the epoch that is both finalized and reportable
+    current_frame = oracle.functions.getCurrentFrame().call()
+    potentially_reportable_epoch = current_frame[0]
+    logging.info(f'Potentially reportable epoch: {potentially_reportable_epoch} (from ETH1 contract)')
+    finalized_epoch_beacon = beacon.get_finalized_epoch()
+    # For Web3 client
+    # finalized_epoch_beacon = int(beacon.get_finality_checkpoint()['data']['finalized']['epoch'])
+    logging.info(f'Last finalized epoch: {finalized_epoch_beacon} (from Beacon)')
+    partial_metrics.epoch = min(potentially_reportable_epoch,
+                                (finalized_epoch_beacon // epochs_per_frame) * epochs_per_frame)
+    partial_metrics.timestamp = get_timestamp_by_epoch(beacon_spec, partial_metrics.epoch)
+    partial_metrics.depositedValidators = pool.functions.getBeaconStat().call()[0]
+    partial_metrics.bufferedBalance = pool.functions.getBufferedEther().call()
+    return partial_metrics
+
+
+def get_full_current_metrics(w3, beacon, beacon_spec, partial_metrics) -> PoolMetrics:
     """If partial result provided, the oracle fetches all the required states from ETH1 and ETH2"""
```

Comment:
If a(article) partial result IS(passive voice) provided, the oracle fetches all the required states from ETH1 and ETH2

Reply:
Fixed